src/providers/aws.rs

```diff
@@ -1327,22 +1351,35 @@ impl RegionLauncher {
                                 tracing::trace!("instances not yet shut down -- retrying");
                                 tokio::time::delay_for(tokio::time::Duration::from_secs(5)).await;
                             } else {
-                                tracing::warn!("failed: {}", err);
-                                break;
+                                Err(Report::msg(err.to_string())
```

Comment:
```suggestion
                                Err(eyre!(RusotoError::Unkown(r))
```

Reply:
That way we keep the full underlying error, rather than just its message.

Reply:
From https://docs.rs/eyre/0.4.2/eyre/macro.eyre.html?search= it seems like `eyre!` just uses its argument as a `Debug + Display` anyway? Is there anything extra actually being kept?

Reply:
Doing this results in the following error:
```
error[E0277]: the trait bound `eyre::Report: std::error::Error` is not satisfied
    --> src/providers/aws.rs:1360:94
     |
1360 | ...                   .wrap_err("failed to clean up temporary security group"))?;
     |                                                                                ^ the trait `std::error::Error` is not implemented for `eyre::Report`
     |
     = note: required because of the requirements on the impl of `std::convert::From<eyre::Report>` for `eyre::Report<color_eyre::Context>`
     = note: required by `std::convert::From::from`
```

Reply:
@yaahc this seems like an odd error. any idea what causes it?

Reply:
It seems to be that the wrapped error does not implement `std::error::Error`, though the error message is then pretty misleading :'(

Reply:
Although it _looks_ like it does: https://docs.rs/rusoto_core/0.43.0/rusoto_core/enum.RusotoError.html#impl-Error + https://docs.rs/rusoto_ec2/0.43.0/rusoto_ec2/enum.TerminateInstancesError.html#impl-Error.